Resolve Export Coupon Code Grid redirect to DashBoard when create New Cart Price Rule issue24468

edenduong opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

Coupon Grid in create new "Cart Price Rule" should be disabled. Because it is useless and when exporting, it has the error

Solution: Hide the grid when create new 

1. Resolve magento/magento2#24468: Export Coupon Code Grid redirect to DashBoard when create New Cart Price Rule

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#24468: Export Coupon Code Grid redirect to DashBoard when create New Cart Price Rule

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Go to backend
2. Marketing-> Cart Price Rules, Create New Cart Price Rule
3. In " Coupon" choose Specific Coupon
4. Check "Use Auto Generation"
5. Scroll to Manage Coupon Codes
6. Export

#### Expected result (*)
<!--- Tell us what do you expect to happen. -->
1. Should export empty file

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

edenduong commented:
No introduce the Public function, because it already in the Parent Class

dmytro-ch commented on app/code/Magento/SalesRule/Block/Adminhtml/Promo/Quote/Edit/Tab/Coupons/Grid.php:
```diff
@@ -151,4 +152,16 @@ public function getGridUrl()
+    /**
+     * @inheritdoc
+     */
+    public function toHtml()
```
@edenduong, could you please verify if it's possible to solve the mentioned issue by overriding the `_toHtml` method instead of `toHtml`?
If not, then let's leave it as it is.

Thank you!

edenduong replied:
Hello @dmytro-ch ,

I have changed the source code to _toHtml(). Please check it again.

Thanks

magento-engcom-team commented:
Hi @dmytro-ch, thank you for the review.
[ENGCOM-5822](https://jira.corp.magento.com/browse/ENGCOM-5822) has been created to process this Pull Request
:eight_spoked_asterisk: @dmytro-ch, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

edenduong commented:
@magento give me test instance

magento-engcom-team commented:
Hi @edenduong. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @edenduong, here is your [new Magento instance](https://pr-24471.instances.magento-community.engineering).
Admin access: [https://pr-24471.instances.magento-community.engineering/admin](https://pr-24471.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

edenduong commented:
Hi @engcom-Alfa, @dmytro-ch : I fixed it. Please check it again.

magento-engcom-team commented:
Hi @sidolov, thank you for the review.
[ENGCOM-5822](https://jira.corp.magento.com/browse/ENGCOM-5822) has been created to process this Pull Request

engcom-Alfa commented:
:heavy_check_mark: QA Passed

sidolov reviewed (changes requested):
@edenduong after a closer look, we found that it would be better disable Manage Coupon Codes grid controls or fix Export button functionality to handle cases when the grid is empty.

VladimirZaets commented:
@edenduong can you take a look at it?

edenduong committed:
Resolve Export Coupon Code Grid redirect to DashBoard when create New Cart Price Rule issue24468

edenduong commented:
Hi @VladimirZaets : I have changed the source code. I fixed Export button functionality to handle cases when the grid is empty instead of hide the grid.

edenduong committed:
Fix static test

slavvka commented:
@magento give me test instance

magento-engcom-team commented:
Hi @slavvka. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @slavvka, here is your [new Magento instance](https://pr-24471.instances.magento-community.engineering).
Admin access: [https://pr-24471.instances.magento-community.engineering/admin](https://pr-24471.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

magento-engcom-team commented:
Hi @dmytro-ch, thank you for the review.
[ENGCOM-5822](https://jira.corp.magento.com/browse/ENGCOM-5822) has been created to process this Pull Request

magento-engcom-team commented:
Hi @slavvka, thank you for the review.
[ENGCOM-5822](https://jira.corp.magento.com/browse/ENGCOM-5822) has been created to process this Pull Request

engcom-Alfa commented:
:heavy_check_mark: QA Passed
